### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by duanemay**
Mostly automated with Openrewrite rules, and some manual updates.

Rules Run: CodeStyle, Common Static Analysis Issue remediation, Java 17

### CODE REVIEW

File: model/src/main/java/org/cloudfoundry/identity/uaa/codestore/ExpiringCode.java [hunk 1]
```diff
@@ -76,28 +76,30 @@ public void setIntent(String intent) {
...
+        if (!Objects.equals(code, that.code)) {
             return false;
-        if (!Objects.equals(expiresAt, that.expiresAt))
+        }
+        if (!Objects.equals(data, that.data)) {
             return false;
-
-        return true;
+        }
+        return !!Objects.equals(expiresAt, that.expiresAt);
```
Comment (strehle) [inline model/src/main/java/org/cloudfoundry/identity/uaa/codestore/ExpiringCode.java]: indendet to have !! ?

**Review by strehle — COMMENTED** [review]
can you please check these 7 findings about
!!

https://sonarcloud.io/project/issues?severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&types=BUG&pullRequest=3174&id=cloudfoundry-identity-parent

File: model/src/main/java/org/cloudfoundry/identity/uaa/codestore/ExpiringCode.java (hunk 1 above)
Comment (duanemay) [reply model/src/main/java/org/cloudfoundry/identity/uaa/codestore/ExpiringCode.java]: Part of using automated tools... Fixed

File: model/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/SamlServiceProvider.java [hunk 2]
```diff
@@ -149,58 +149,71 @@ public boolean configIsValid() {
...
                 return false;
-        } else if (!entityId.equals(other.entityId))
-            return false;
-        if (version != other.version)
+            }
+        } else if (!entityId.equals(other.entityId)) {
             return false;
-        return true;
+        }
+        return !(version != other.version);
```
Comment (strehle) [inline model/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/SamlServiceProvider.java]: See here logic from your tool or was there before, but can be simplified

**Review by strehle — COMMENTED** [review]
Please resolve with develop

File: model/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/SamlServiceProvider.java (hunk 2 above)
Comment (duanemay) [reply model/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/idp/SamlServiceProvider.java]: previous was
``` 
  if (version != other.version)
            return false;
        return true;
``` 

I rebased develop, and added a few more fixes, now:
```
        return version == other.version;
```

**Comment by strehle** [comment]
@duanemay So far I am through with, thanks for the cleanup. I have 2 open clarifications

1. The test coverage is low because some objects from IdP removal ( see https://github.com/cloudfoundry/uaa/pull/2638 , reason of major upgrade to 77.x ) still there. Suggest to remove these objects with an extra PR, WDYT ?

2. Guava. Should we remove Guava from usage at all ? Because right now you removed the direct dependency only but then we get Guava from the transient dependency from opensaml 4.x . We can exclude it in general but then also some other classes still use Util functions from Guava.  Problem now: transient dependency adds Guava 28.x which has some CVEs. With that package we run into a problem with our validation

**Comment by strehle** [comment]
> @duanemay So far I am through with, thanks for the cleanup. I have 2 open clarifications
> 
> 1. The test coverage is low because some objects from IdP removal ( see [Remove SAML IDP feature #2638](https://github.com/cloudfoundry/uaa/pull/2638) , reason of major upgrade to 77.x ) still there. Suggest to remove these objects with an extra PR, WDYT ?

Tested without guava and yes, 
`net.shibboleth.utilities.java.support.xml.BasicParserPool.setBuilderFeatures`
needs it, so we cannot omit but then it is better we define the version of used guava and use the lastest. Otherwise we get an outdated version if we rely on transitive deps.

**Review by strehle — APPROVED** [review]

**Comment by mikeroda** [comment]
@duanemay, can you give the specific rewrite recipes that were used for this code cleanup?

**Pull request merged**